packages/firestore/test/unit/model/target_index_matcher.test.ts

```diff
@@ -0,0 +1,687 @@
+    );
+
+    for (const query of queriesWithEqualitiesAndAscendingOrder) {
+      validateServesTarget(query, 'a', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'b', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'a', IndexKind.CONTAINS);
+    }
+  });
+
+  it('equalities with descending order', () => {
+    const queriesWithEqualitiesAndDescendingOrder = queriesWithEqualities.map(
+      q => queryWithAddedOrderBy(q, orderBy('a', 'desc'))
+    );
+
+    for (const query of queriesWithEqualitiesAndDescendingOrder) {
+      validateServesTarget(query, 'a', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'b', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'a', IndexKind.CONTAINS);
+    }
+  });
+
+  it('inequalitiesWithDefaultOrder', () => {
+    for (const query of queriesWithInequalities) {
+      validateServesTarget(query, 'a', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'b', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'a', IndexKind.CONTAINS);
+    }
+  });
+
+  it('inequalitiesWithAscendingOrder', () => {
+    const queriesWithInequalitiesAndAscendingOrder =
+      queriesWithInequalities.map(q =>
+        queryWithAddedOrderBy(q, orderBy('a', 'asc'))
+      );
+
+    for (const query of queriesWithInequalitiesAndAscendingOrder) {
+      validateServesTarget(query, 'a', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'b', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'a', IndexKind.CONTAINS);
+    }
+  });
+
+  it('inequalitiesWithDescendingOrder', () => {
+    const queriesWithInequalitiesAndDescendingOrder =
+      queriesWithInequalities.map(q =>
+        queryWithAddedOrderBy(q, orderBy('a', 'desc'))
+      );
+
+    for (const query of queriesWithInequalitiesAndDescendingOrder) {
+      validateServesTarget(query, 'a', IndexKind.DESCENDING);
+      validateDoesNotServeTarget(query, 'b', IndexKind.ASCENDING);
+      validateDoesNotServeTarget(query, 'a', IndexKind.CONTAINS);
+    }
+  });
+
+  it('inequality uses single field index', () => {
+    const q = queryWithAddedFilter(
+      queryWithAddedFilter(query('collId'), filter('a', '>', 1)),
+      filter('a', '<', 10)
+    );
+    validateServesTarget(q, 'a', IndexKind.ASCENDING);
+  });
+
+  it('is query uses merge join', () => {
+    const q = queryWithAddedFilter(
+      queryWithAddedFilter(query('collId'), filter('a', 'in', [1, 2])),
+      filter('b', '==', 5)
+    );
+    validateServesTarget(q, 'a', IndexKind.ASCENDING);
+    validateServesTarget(q, 'b', IndexKind.ASCENDING);
+    validateServesTarget(q, 'a', IndexKind.ASCENDING, 'b', IndexKind.ASCENDING);
+  });
+
+  it('validates collection', () => {
+    {
+      const targetIndexMatcher = new TargetIndexMatcher(
+        queryToTarget(query('collId'))
+      );
+      const index = fieldIndex('collId');
+      expect(() => targetIndexMatcher.servedByIndex(index)).to.not.throw;
```

Comment:
I don't see any `throw` statements in target_index_matcher.ts. Similarly, I don't see any throw statements in TargetIndexMatcher.java.

does `.to.throw` and `.to.not.throw` apply to assertions (e.g. the `debugAssert` in the code)? The other examples of `.to.throw` that I found in other test files corresponded to a `throw` statement in the source.

Reply:
Yes it applies to the assert.